docs/Developer Documentation/ResumeController/index.md

```diff
@@ -0,0 +1,237 @@
+
+If some of the data failed to restore, `ResumptionDataProcessor` will revert already restored data and call  `callback(ERROR_CODE, info)`.
+
+The requirements are available in proposal 0190: [Handle response from HMI during resumption data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md)
+
+RegisterAppInterface will wait for the callback to send a response to a mobile application.
+
+### AppExtension
+
+Application extension contains follwowing methods for resumption : 
+```cpp
+  /**
+   * @brief SaveResumptionData method called by SDL when it saves resumption
+   * data.
+   * @param resumption_data data reference to data, that will be appended by
+   * plugin
+   */
+  virtual void SaveResumptionData(
+      smart_objects::SmartObject& resumption_data) = 0;
+
+  /**
+   * @brief ProcessResumption Method called by SDL during resumption.
+   * @param resumption_data list of resumption data
+   * @param subscriber callbacks for subscribing
+   */
+  virtual void ProcessResumption(
+      const smart_objects::SmartObject& resumption_data,
+      resumption::Subscriber subscriber) = 0;
+
+  /**
+   * @brief RevertResumption Method called by SDL during revert resumption.
+   * @param subscriptions Subscriptions from which must discard
+   */
+  virtual void RevertResumption(
+      const smart_objects::SmartObject& subscriptions) = 0;
+```
+
+Only an application's extension have an access to active data, data send and data revert process. 
+Each application extension uses its own plugin to manipulate with functionality. 
+
+`SaveResumptionData` will fill passed `resumption_data` for saving to `ResumptionData`. 
+
+
+Example from VehicleInfoAppExtension: 
+```cpp
+SDLRPCPlugin& plugin_;
+...
+void VehicleInfoAppExtension::SaveResumptionData(
+    smart_objects::SmartObject& resumption_data) {
+  resumption_data[strings::application_vehicle_info] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  int i = 0;
+  for (const auto& subscription : subscribed_data_) {
+    resumption_data[strings::application_vehicle_info][i++] = subscription;
+  }
+}
+```
+
+
+`ProcessResumption` will send appropriate HMI requests, and change internal SDL state according to provided `resumption_data`. All HMI responses will be transferred to `subscriber`.
+
+
+Example from SDLAppExtension: 
+```cpp
+SDLRPCPlugin& plugin_;
+...
+void SDLAppExtension::ProcessResumption(
+    const smart_objects::SmartObject& saved_app,
+    resumption::Subscriber subscriber) {
+  ...
+  const bool subscribed_for_way_points =
+      saved_app[strings::subscribed_for_way_points].asBool();
+  if (subscribed_for_way_points_so) {
+    plugin_.ProcessResumptionSubscription(app_, *this, subscriber);
+  }
+}
+```
+
+`resumption::Subscriber` is the function that will notify ResumptionDataProcessor about request sent to HMI by plugin in scope of resumption.
```

Comment:
note that it was requested subscriber be changed to an interface method like `subscribeToResponse` on ResumptionDataProcessor to be called by clients instead

Reply:
Fixed 067a963 
The core will be also changed